xdsl/dialects/gpu.py

```diff
@@ -342,12 +339,12 @@ class ModuleOp(IRDLOperation):
 
     traits = frozenset([IsolatedFromAbove()])
 
-    @staticmethod
-    def get(name: SymbolRefAttr, ops: Sequence[Operation]) -> ModuleOp:
-        op = ModuleOp.build(attributes={"sym_name": name}, regions=[ops])
-        return op
+    def __init__(self, name: SymbolRefAttr, ops: Sequence[Operation]):
+        return super().__init__(attributes={"sym_name": name}, regions=[ops])
 
-    def verify_(self):
+    def verify_(
+        self,
+    ):
         if not isinstance(self.body.block.last_op, ModuleEndOp):
```

Comment:
Regarding this requirement of having a `module_end` operation here; that's not the case as I understand it in MLIR currently, right? However, in MLIR this operation implements the `SingleBlockImplicitTerminator` trait which allows omitting the terminator. Do I understand this right? I'm asking because I was looking at it earlier today :)

Reply:
yeah, exactly. Sorry, I missed your reply before the merge, but I was looking in MLIR and I couldn't find a single use of this terminator (filechecks, etc.).
So, not sure what is the point.

I was going to suggest leaving TODO note/issue regarding this for if/when we implement the `SingleBlockImplicitTerminator`, but I can do it in a supplementary PR.